core/unit_test/TestTeamCombinedReducers.hpp

```diff
@@ -0,0 +1,478 @@
+              Kokkos::Min<int>(teamResult2), Kokkos::Max<int>(teamResult3));
+
+          Kokkos::single(Kokkos::PerTeam(team), [=]() {
+            teamView(0) += teamResult0;
+            teamView(1) += teamResult1;
+            teamView(2) += teamResult2;
+            teamView(3) += teamResult3;
+          });
+        });
+
+    auto hostView = Kokkos::create_mirror_view_and_copy(
+        Kokkos::DefaultHostExecutionSpace(), teamView);
+
+    EXPECT_EQ((n * (n + 1) / 2), hostView(0));
+    EXPECT_EQ(std::pow(n, n), hostView(1));
+    EXPECT_EQ(1, hostView(2));
+    EXPECT_EQ(n, hostView(3));
+  }
+
+  void test_team_vector_range_combined_reducers(const int n) {
+    auto policy = Kokkos::TeamPolicy<TEST_EXECSPACE>(1, Kokkos::AUTO);
+    using team_member_type = decltype(policy)::member_type;
+
+    auto teamView = Kokkos::View<int*, TEST_EXECSPACE::memory_space>("view", 4);
+
+    Kokkos::parallel_for(
+        policy, KOKKOS_LAMBDA(team_member_type const& team) {
+          auto teamVectorRange = Kokkos::TeamVectorRange(team, n);
+          int teamResult0      = 0;
+          int teamResult1      = 0;
+          int teamResult2      = 0;
+          int teamResult3      = 0;
+
+          Kokkos::parallel_reduce(
+              teamVectorRange,
+              [=](int const& i, int& localVal0, int& localVal1, int& localVal2,
+                  int& localVal3) {
+                localVal0 += i + 1;
+                localVal1 += i + 1;
+                localVal2 = (localVal2 < (i + 1)) ? (i + 1) : localVal2;
+                localVal3 += n;
+              },
+              teamResult0, Kokkos::Sum<int>(teamResult1),
+              Kokkos::Max<int>(teamResult2), teamResult3);
+
+          Kokkos::single(Kokkos::PerTeam(team), [=]() {
+            teamView(0) += teamResult0;
+            teamView(1) += teamResult1;
+            teamView(2) += teamResult2;
+            teamView(3) += teamResult3;
+          });
+        });
+
+    auto hostView = Kokkos::create_mirror_view_and_copy(
+        Kokkos::DefaultHostExecutionSpace(), teamView);
+
+    EXPECT_EQ((n * (n + 1) / 2), hostView(0));
+    EXPECT_EQ((n * (n + 1) / 2), hostView(1));
+    EXPECT_EQ(n, hostView(2));
+    EXPECT_EQ(n * n, hostView(3));
+  }
+};
+
+TEST(TEST_CATEGORY, team_thread_range_combined_reducers) {
+#if defined(KOKKOS_ENABLE_OPENMPTARGET)
+  if constexpr (std::is_same_v<TEST_EXECSPACE,
+                               Kokkos::Experimental::OpenMPTarget>)
+    GTEST_SKIP() << "team_reduce with a generic reducer is not implemented for "
+                 << TEST_EXECSPACE::name();
+
+#elif defined(KOKKOS_ENABLE_OPENACC)
+  if constexpr (std::is_same_v<TEST_EXECSPACE, Kokkos::Experimental::OpenACC>)
+    GTEST_SKIP() << "team_reduce with a generic reducer is not implemented for "
+                 << TEST_EXECSPACE::name();
+#endif
+
+  TeamTeamCombinedReducer tester;
+  tester.test_team_thread_range_only_scalars(5);
```

Comment:
Would there be any value in testing also with base case `n=0` for at least one of the cases?

Reply:
Good point. I added in base case of `n=0` for each team policy and found some deficiencies in these tests.
It turns out that when `n=0` and when the inner `parallel_reduce` do not run, the outer `parallel_for` loop still runs, initializing intermediate team reduced values based on the corresponding reducer.
Added in if conditions to avoid unnecessary aggregates when the inner `parallel_reduce` do not run.